docs/source/topics/data_types.rst

```diff
@@ -0,0 +1,508 @@
+       [-1.36769345, -1.24523403,  2.94149041,  3.6004033 ,  6.98809593],
+       [-1.42050683, -1.12604118,  2.48497007,  3.69389815,  7.07537154],
+       [-1.52788845, -0.95900776,  2.09104321,  3.82330632,  7.20537566],
+       [-1.71354964, -0.74425095,  1.82242466,  3.98697455,  7.37979746]])
+
+To insert kpoints and bands in the ``bs`` object we should employ ``set_kpoints()`` and ``set_bands()`` methods:
+
+.. code-block:: python
+
+    bs.set_kpoints(kpoints)
+    bs.set_bands(bands, units='eV')
+
+    bs.show_mpl() # to visualize the bands
+
+From now the band structure can be visualized.
+Last thing that we may want to add is the array of kpoint labels:
+
+.. code-block:: python
+
+    labels = [(0, 'GAMMA'),
+              (5, 'X'),
+              (6, 'X'),
+              (11, 'U')]
+
+    bs.labels = labels
+    bs.show_mpl() # to visualize the bands
+
+The resulting band structure will look as follows
+
+.. figure:: include/bands.png
+
+.. warning:: Once the ``bs`` object is stored (``bs.store()``) -- it won't accept any modifications.
+
+.. _topics:data_types:materials:bands:plotting:
+
+Plotting the band structure
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may notice that depending on how you assign the kpoints labels the output of the ``show_mpl()`` method looks different.
+Please compare:
+
+.. code-block:: python
+
+    bs.labels = [(0, 'GAMMA'),
+              (5, 'X'),
+              (6, 'Y'),
+              (11, 'U')]
+    bs.show_mpl()
+
+    bs.labels = [(0, 'GAMMA'),
+              (5, 'X'),
+              (7, 'Y'),
+              (11, 'U')]
+    bs.show_mpl()
+
+In the first case two neighboring kpoints with ``X`` and ``Y`` labels will look like ``X|Y``, while in the second case they will be separated by a certain distance.
+The logic behind such a difference is the following.
+In the first case the plotting method discovers the two neighboring kpoints and assumes them to be a discontinuity point in the band structure (e.g. Gamma-X|Y-U).
+In the second case the kpoints labelled ``X`` and ``Y`` are not neighbors anymore, so they are
+plotted with a certain distance between them.
+The intervals between the kpoints on the X axis are proportional to the cartesian distance between them.
+
+.. _topics:data_types:materials:bands:spins:
+
+Dealing with spins
+^^^^^^^^^^^^^^^^^^
+
+The ``BandsData`` object can also deal with the results of spin-polarized calculations.
+Two provide different bands for two different spins you should just merge them in one array and import them again using the ``set_bands()`` method:
+
+.. code-block:: python
+
+    bands_spins = [bands, bands-0.3] # to distinguish the bands of different spins we substitute 0.3 from the second band structure
+    bs.set_bands(bands_spins, units='eV')
+    bs.show_mpl()
+
+Now the shape of the bands array becomes ``nspins * nkpoints * nstates``
+
+.. _topics:data_types:export:
+
+Export data nodes to various formats
```

Comment:
As you already mentioned, maybe all this section should be moved to an 'exporting data' section - like in "importing", we have two types (import an AiiDA DB, and import from external sources). Also here, we might want to have two types of export: export to AiiDA export file, and export to other formats (this section for the AiiDA data types).

Reply:
So far I have moved the CLI examples to the corresponding data type section (as they are rather specific) and kept the general description of the `export()` method at the end of the file. I suppose it could be fit into the "How to share data" section?

Reply:
Yes, but I think we also discussed the option to just leave them at the bottom of the page in yesterday's meeting - I'm not sure which solution is better